docs/rfc/rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,157 @@
+blocks member with index 2 tries to submit, after 12 blocks member with index 3
+tries to submit and so forth.
+
+Anyone should be able to submit a transaction creating a new group and this
+action should be detached from relay entry submission. The submitter of this
+transaction receives a part of the group creation pool and the remaining part
+should be reserved for DKG result submitter as reimbursement for future
+transaction. What percentage of the pool goes where is a governable parameter
+and should be determined based on the gas cost of the two transactions.
+
+To make this scheme works both relay entry and DKG result submission
+transactions need to have predictable gas cost. Although it is guaranteed for
+the latter, the former, in v1 of the beacon, depends on the gas cost of
+executing a callback and needs to be optimized.
+
+T rewards will be distributed continuously to all operators in the beacon
+sortition pool, just like in the case of tBTC v2.
+
+=== Callbacks
+
+In v1 of the random beacon, a callback is executed in the same transaction in
+which relay entry is submitted with a gas limit of 2M. This allows applications
+using the random beacon to avoid an additional complexity at their end but comes
+at the cost of random beacon operators who need to have enough ETH on their
+balance. This approach is not compatible with the idea that all rewards will be
+solely in T given that it is impossible to establish the cost of executing a
+callback in T.
+
+Instead of executing any callback in the same transaction in which relay entry
+is submitted, we can assume that only the simplest callbacks are allowed, up to
+some gas limit set by the governance. Applications wanting to use a relay entry
+should submit another transaction using the relay entry value previously set by
+the random beacon. They should also employ an additional security check making
+sure the entry submitted is only valid for a certain number of blocks to avoid
+situation when relay entry beacon transaction and application-specific
+transaction are executed far from each other.
+
+On the example of tBTC v2, random beacon transaction submitting relay entry
+would only execute a simple callback setting the value of the entry on wallet
+factory, and another transaction executed on tBTC v2 side to the wallet factory,
+would use the entry for sortition.
+
+The fact the full - even the most complex - callback is executed in the same
+transaction in which the relay entry is submitted gives an impression of better
+security. This impression is false though, given that the entry to be submitted
+is visible in the mempool and smart attackers can have their transactions mined
+faster to - for example - put the sortition pool in the desired state for the
+new entry. This issue needs to be solved on the sortition pool side with
+initiation time for new operators in the pool.
+
+=== Group creation
+
+As mentioned in the section about fees and rewards, anyone should be able to
```

Comment:
Can you hyperlink the fees and rewards section?

Reply:
https://github.com/keep-network/keep-core/pull/2585/commits/2a0d1e1c217328743c47a0b1363a3e19c10e170e